src/trusted_ca_keys/trusted_ca_keys.cc

```diff
@@ -0,0 +1,161 @@
+
+using oslogin_utils::HttpGet;
+using oslogin_utils::ParseJsonToSuccess;
+using oslogin_utils::ParseJsonToEmail;
+using oslogin_utils::ParseJsonToSshKeys;
+using oslogin_utils::UrlEncode;
+using oslogin_utils::ParseJsonToCAKeys;
+
+ofstream
+CreateTmpKeysFile(const string& trusted_keys_filename, const string& trusted_keys_tmp_filename) {
+
+  ofstream trusted_keys_tmp_file(trusted_keys_tmp_filename.c_str());
+
+  if (!trusted_keys_tmp_file.is_open()) {
+    stringstream err_msg;
+
+    err_msg << "Failed!\n[ERROR] Can't open '"
+            << trusted_keys_tmp_filename << "' for writing";
+    perror(err_msg.str().c_str());
+
+    exit(EXIT_FAILURE);
+  }
+
+  if (chown(trusted_keys_tmp_filename.c_str(), 0, 0) != 0) {
+    stringstream err_msg;
+
+    err_msg << "Failed!\n[ERROR] Failed to set root:root file owner for "
+            << trusted_keys_tmp_filename;
+    perror(err_msg.str().c_str());
+    remove(trusted_keys_tmp_filename.c_str());
+
+    exit(EXIT_FAILURE);
+  }
+
+  if (chmod(trusted_keys_tmp_filename.c_str(),
+            S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH) != 0) {
+    stringstream err_msg;
+
+    err_msg << "Failed!\n[ERROR] Failed to set 644 file permissions for "
+            << trusted_keys_tmp_filename;
+    perror(err_msg.str().c_str());
+    remove(trusted_keys_tmp_filename.c_str());
+
+    exit(EXIT_FAILURE);
+  }
+
+  return trusted_keys_tmp_file;
+}
+
+void
+WriteCAKeysTmpFile(ofstream& trusted_keys_tmp_file, const vector<string> ca_keys) {
+
+  for (size_t i = 0; i < ca_keys.size(); ++i) {
+    trusted_keys_tmp_file << ca_keys[i] << endl;
+  }
+}
+
+int
+main(int argc, char* argv[]) {
+
+  string trusted_keys_filename = trusted_keys_file_def;
+  string mds_url = mds_url_def;
+  char*  mds_url_env = NULL;
+
+  if (argc >= 2) {
+    if (strncmp(argv[1], "--help", 6) == 0 ||
+        strncmp(argv[1], "-h", 2) == 0) {
+      cerr << "Usage: google_trusted_ca_keys [/path/to/trustedca.pub]\n\n"
+           << "Environment:\n"
+           << " - GOOGLE_OSLOGIN_MDS_URL=\"http://srv/path/to/certificates\"\n"
+           << "Default values:\n"
+           << " - CA keys file: /etc/ssh/trustedca.pub\n"
+           << " - API endpoint: http://169.254.169.254/computeMetadata/v1/oslogin/certificates\n";
+
+      return EXIT_FAILURE;
+    }
+    trusted_keys_filename = argv[1];
+  }
+
+  if ( (mds_url_env = getenv("GOOGLE_OSLOGIN_MDS_URL")) != NULL) {
+    mds_url = mds_url_env;
+  }
+
+  string trusted_keys_tmp_filename = trusted_keys_filename + tmp_suff;
+  cout << "Updating the Trusted CA keys file... " << flush;
+  ofstream trusted_keys_tmp_file = CreateTmpKeysFile(trusted_keys_filename,
+                                                     trusted_keys_tmp_filename);
+
+  string user_response;
+  long http_code = 0;
+  if (!HttpGet(string(mds_url.c_str()), &user_response, &http_code) ||
+      user_response.empty() || http_code != 200) {
+
+    cerr << "Failed!\n[ERROR] API call has failed. Metadata API endpoint '"
```

Comment:
don't you want to remove/close the temp file here?

Reply:
Good catch! 
Fixed.